### PR TITLE
Switch to debops-ci

### PR DESCRIPTION
**Pull request by FedericoCeratto**
Simplify the CircleCI conf, add a more friendly tool.
- Detect package directories
- Bump up version number in PR builds
- Support manual runs from CLI on workstations
- Show VCS changes in package changelog e.g. https://github.com/ooni/pipeline/compare/d68112c8...56d0ac6b
 
[Please squash commits]

### CODE REVIEW

File: .circleci/mkci [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+
+def upload(fi, args) -> None:
+    """Upload to Bintray"""
+    assert fi.is_file()
+    pname, pver, arch = fi.name.split("_")
+    btuser = getenv("BINTRAY_USERNAME")
+    assert btuser
+    auth = HTTPBasicAuth(btuser, getenv("BINTRAY_API_KEY"))
+    url = f"{BINTRAY_API}/content/{args.org}/{args.repo}/{pname}/{pver}/{fi.name};deb_distribution=unstable;deb_component=main;deb_architecture=amd64;publish=1"
```
Comment (hellais) [inline .circleci/mkci]: I wonder if you want to maybe add a version check at the beginning of the script to ensure it's python > 3.7

Comment (hellais) [reply .circleci/mkci]: Also maybe wrap this line

File: .circleci/config.yml [hunk 2]
```diff
@@ -5,32 +5,8 @@ jobs:
...
-      #     key: foo-0001
-      #     paths:
-      #       - ~/cache
-      - store_artifacts:
-          path: /tmp/debs/
-          destination: debs
-      - store_test_results:
-          path: /tmp/debs/
+            apt-get install -y --no-install-recommends git python3 python3-requests
+            ./.circleci/mkci ci
```
Comment (hellais) [inline .circleci/config.yml]: Maybe to avoid confusion with `measurement-kit` the script can be renamed `build-package`?

File: .circleci/mkci [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    sudo = True
+    if getenv("CIRCLECI"):
+        # Running in CircleCI
+        sudo = False
+        comp = trim_compare(getenv("CIRCLE_COMPARE_URL", ""))  # show changes in VCS
+        if getenv("CIRCLE_PULL_REQUEST"):
+            # This is a PR: build ~pr<N>-<N> version. CIRCLE_PR_NUMBER is broken
+            pr_num = getenv("CIRCLE_PULL_REQUEST", "").rsplit("/", 1)[-1]
+            build_num = getenv("CIRCLE_BUILD_NUM")
+            ver = f"{ver}~pr{pr_num}-{build_num}"
```
Comment (hellais) [inline .circleci/mkci]: Maybe this is something which can go in a dedicated function or higher up in the script as a constant?

**Review by hellais — COMMENTED** [review]
I left a few nitpick comments. My knowledge of linux packaging and ability to test it is limited, but it looks reasonable to me.

**Comment by FedericoCeratto** [comment]
Usage example:
```
sudo cat /etc/apt/sources.list.d/ooni-internal.list
deb [trusted=yes] https://dl.bintray.com/ooni/internal-pull-requests unstable main

sudo apt-get update
sudo apt-get install fastpath
apt-cache show fastpath | grep Ver
dpkg -L fastpath  # show installed packages
sudo apt-get install fastpath=0.7  # rollback
```

**Comment by FedericoCeratto** [comment]
Also, the changelog of the installed package links to the code changes:
```
zcat /usr/share/doc/detector/changelog.Debian.gz
...
https://github.com/ooni/pipeline/compare/22118e60...4818169b
```

```

File: .circleci/mkci (hunk 1 above)
Comment (FedericoCeratto) [reply .circleci/mkci]: Done

File: .circleci/config.yml (hunk 2 above)
Comment (FedericoCeratto) [reply .circleci/config.yml]: Naming things is notoriously difficult and now it's doing CI stuff as well...

File: .circleci/mkci (hunk 3 above)
Comment (FedericoCeratto) [reply .circleci/mkci]: I'm planning of moving the CircleCI and Bintray-specific code out in future to make the "core" service-independent.

**Comment by hellais** [comment]
LGTM

**Pull request merged**